### PR TITLE
Updated "tf.keras.losses.cosine_similarity()" docs

### LINKED ISSUE #37896: tf.keras.losses.cosine_similarity() is not a negative quantity
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/losses/cosine_similarity

## Description of issue (what needs changing):
Documentation states that tf.keras.losses.cosine_similarity() "is a negative quantity between -1 and 0, where 0 indicates orthogonality and values closer to -1 indicate greater similarity."
But it is actually not true. tf.keras.losses.cosine_similarity() can return positive values.

### Usage example
```
>>> import tensorflow as tf
>>> tf.keras.losses.cosine_similarity([[1., 1.]], [[-1., -1.]])
<tf.Tensor: shape=(1,), dtype=float32, numpy=array([0.99999994], dtype=float32)>
```

### PR DESCRIPTION
**Pull request by ManishAradwad**
Fixes : #37896 
@mihaimaruseac Could you plz review this?

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
This is not true. Cosine similarity will never be `pi` for example, but the docstring rewrite suggests that as a possibility.

The scalar product of two unit norm vectors is between -1 and 1. However, the way this similarity is used in optimization requires the output to be -1 and 0.

Adding @pavithrasv and @fchollet for discussion between changing only documentation so that it is between -1 and 1 or changing code to have output still be between -1 and 0.

**Comment by mihaimaruseac** [comment]
Let's only update docstring to say output is betwene -1 and 1

**Comment by ManishAradwad** [comment]
@mihaimaruseac Can you plz see if the docstring is fine??

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
You should only replace 0 to -1 in the first line.

Then -1 is similarity, 0 is orthogonality, 1 is totally opposite values

I would also modify the doctest below to include a case where output gets close to 1.

**Comment by ManishAradwad** [comment]
@mihaimaruseac I've added the statement for values closer to 1 as you've suggested. I had to make some changes after that to make sure lines didn't get too long. All other statements are same as before. Plz review it.

**Comment by mihaimaruseac** [comment]
> I would also modify the doctest below to include a case where output gets close to 1.

LGTM, but please don't forget about this part

File: tensorflow/python/keras/losses.py [hunk 1]
```diff
@@ -1683,10 +1683,11 @@ def cosine_similarity(y_true, y_pred, axis=-1):
 
   Note that it is a number between -1 and 1. When it is a negative number between
   -1 and 0, 0 indicates orthogonality and values closer to -1 indicate greater
-  similarity. This makes it usable as a loss function in a setting where you try
-  to maximize the proximity between predictions and targets. If either `y_true`
-  or `y_pred` is a zero vector, cosine similarity will be 0 regardless of the
-  proximity between predictions and targets.
+  similarity. The values closer to 1 indicate greater dissimilarity. This makes
```
Comment (ManishAradwad) [inline tensorflow/python/keras/losses.py]: This statement isn't right??

Comment (mihaimaruseac) [reply tensorflow/python/keras/losses.py]: Why?

If the data is `[1, 1]` and `[1, 1]` the vectors are completely similar, the cosine similarity is -1 (dot product is 1, we have a - in front).

If the data is `[1, 1]` and `[-1, -1]` the vectors are opposite each other, the dot product is -1, but due to the `-` in front the metric becomes 1.

If the data is `[0, 1]` and `[1, 0]`, the vectors are orthogonal, and both the dot product as well as the metric value are 0.

Comment (ManishAradwad) [reply tensorflow/python/keras/losses.py]: @mihaimaruseac I'm sorry, I'm a little bit confused.
In the previous comment, I meant doesn't that statement(`The values closer to 1 indicate greater dissimilarity.`) include a case where output gets closer to 1?

Comment (mihaimaruseac) [reply tensorflow/python/keras/losses.py]: I meant that you should include the case where output gets closer to 1 in the doctest example below. The docstring header is good.

File: tensorflow/python/keras/losses.py [hunk 2]
```diff
@@ -1693,16 +1693,16 @@ def cosine_similarity(y_true, y_pred, axis=-1):
...
-  >>> y_pred =[[1., 0.], [1., 1.]]
+  >>> y_true = [[0., 1.], [1., 1.], [1., 1.]]
+  >>> y_pred = [[1., 0.], [1., 1.], [-1., -1.]]
   >>> loss = tf.keras.losses.cosine_similarity(y_true, y_pred, axis=1)
   >>> # l2_norm(y_true) = [[0., 1.], [1./1.414], 1./1.414]]]
   >>> # l2_norm(y_pred) = [[1., 0.], [1./1.414], 1./1.414]]]
   >>> # l2_norm(y_true) . l2_norm(y_pred) = [[0., 0.], [0.5, 0.5]]
   >>> # loss = -sum(l2_norm(y_true) . l2_norm(y_pred), axis=1)
   >>> #       = -[0. + 0., 0.5 + 0.5]
   >>> loss.numpy()
```
Comment (ManishAradwad) [inline tensorflow/python/keras/losses.py]: @mihaimaruseac I've edited the `y_true` and `y_pred` but i couldn't get above lines. I tried getting l2_norms of `y_true` and `y_pred` using `nn.l2_normalize(y_true, axis=1)` and `nn.l2_normalize(y_pred, axis=1)`. I got following output: 
```
tf.Tensor(
[[0.         1.        ]
 [0.70710677 0.70710677]
 [0.70710677 0.70710677]], shape=(3, 2), dtype=float32) tf.Tensor(
[[ 1.          0.        ]
 [ 0.70710677  0.70710677]
 [-0.70710677 -0.70710677]], shape=(3, 2), dtype=float32)
```

Comment (mihaimaruseac) [reply tensorflow/python/keras/losses.py]: 1/1.414 is 1/sqrt(2) is 0.707

I would remove the comment lines

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by ManishAradwad** [comment]
@mihaimaruseac I've removed the comments. Plz review

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@ManishAradwad Can you please address Ubuntu Sanity errors? Thanks!

**Comment by ManishAradwad** [comment]
I went through the errors and it looks like it's failing pylint test. But nothing seems wrong in the docstring. Can somebody plz help?

**Comment by mihaimaruseac** [comment]
Downloading the log shows this near the beginning.

```
FAIL: Found 1 non-whitelisted pylint errors:
tensorflow/python/keras/losses.py:1686: [C0301(line-too-long), ] Line too long (81/80)
```

**Review by mihaimaruseac — APPROVED** [review]

**Review by gbaned — APPROVED** [review]
